File: reverse.js

var processLine = function (line) {
  line = line.trim().split(" ").reverse().join(" ");
  return line;
};

var fs  = require("fs");
fs.readFileSync(process.argv[2]).toString().split('\n').forEach(function (line) {
    if (line != "") {
      var t = processLine(line);
      console.log(t);
    }
});

File: reverse.rb
#!/usr/bin/ruby

file_name = ARGV[0]

if file_name.nil?
	exit
end	

file = File.new(file_name)

while(line = file.gets)
	if line =~ /\w/
		words = line.split(" ").to_a
		words.reverse!
		puts words.join(" ").rstrip
	end
end
